imap_processing/ultra/l1c/ultra_l1c_pset_bins.py

```diff
@@ -359,74 +359,69 @@ def get_deadtime_ratios_by_spin_phase(
-        bins, and the inner arrays contain indices indicating pixels that are below
-        the FWHM scattering threshold.
-    boundary_scale_factors : np.ndarray
+    deadtime_ratios : xarray.DataArray
+        Deadtime ratios at each spin phase step.
+    valid_spun_pixels : xarray.DataArray
+        3D Array of pixels valid at each spin phase step. If rejection based on
+        scattering was set, then these are the pixels below the FWHM scattering
+        threshold and in the field of regard at each spin phase step, and energy
+        shape = (spin_phase_steps, energy_bins, n_pix). IF no rejection,
+        then these are simply the pixels in the field of regard at each spin phase step
+        shape = (spin_phase_steps, 1, n_pix).
+    boundary_scale_factors : xr.DataArray
         Boundary scale factors for each pixel at each spin phase.
-    n_pix : int
-        Number of HEALPix pixels.
+    apply_bsf : bool, optional
+        Whether to apply boundary scale factors. Default is True.
 
     Returns
     -------
     exposure_pointing_adjusted : np.ndarray
         Adjusted exposure times accounting for dead time.
     """
-    # Get energy bin geometric means
-    energy_bin_geometric_means = build_energy_bins()[2]
-    # Exposure time should now be of shape (energy, npix)
-    counts = np.zeros((len(energy_bin_geometric_means), n_pix))
     # nominal spin phase step.
-    nominal_ms_step = 15 / len(pixels_below_scattering)  # time step
+    nominal_ms_step = 15 / valid_spun_pixels.shape[0]  # time step
     # Query the dead-time ratio and apply the nominal exposure time to pixels in the FOR
-    # and below the scattering threshold
-    # Loop through the spin phase steps. This is spinning the spacecraft by nominal
-    # 1 ms steps in the despun frame.
-    for i, pixels_at_spin in enumerate(pixels_below_scattering):
-        # Loop through energy bins
-        for energy_bin_idx in range(len(energy_bin_geometric_means)):
-            pixels_at_energy_and_spin = pixels_at_spin[energy_bin_idx]
-            if pixels_at_energy_and_spin.size == 0:
-                continue
-            # Apply the nominal exposure time (1 ms) scaled by the deadtime ratio to
-            # every pixel in the FOR, that is below the FWHM scattering threshold,
-            counts[energy_bin_idx, pixels_at_energy_and_spin] += (
-                deadtime_ratios[i]
-                * boundary_scale_factors[pixels_at_energy_and_spin, i]
-            )
-
+    # and below the scattering threshold (if scattering rejection is on).
+    # Sum over the first dim of valid_spun_pixels is the spin phase step.
+    # This is like spinning the spacecraft by nominal 1 ms steps in the despun frame.
+    if apply_bsf:
+        counts = (valid_spun_pixels * deadtime_ratios * boundary_scale_factors).sum(
+            dim="spin_phase_step"
+        )
+    else:
+        counts = (valid_spun_pixels * deadtime_ratios).sum(dim="spin_phase_step")
```

Comment:
My personal preference is to extract the final sum since you are doing that over both of these variables. But, either way is fine and you can ignore this if you don't like it.
```suggestion
    all_counts = valid_spun_pixels * deadtime_ratios
    if apply_bsf:
        all_counts *= boundary_scale_factors)
    counts = all_counts.sum(dim="spin_phase_step")
```